indexer/src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt

```diff
@@ -238,48 +177,137 @@ class BackgroundExternalDatabaseSyncingService(
+                createColumns(table, cols, ownerAclKeys)
+            }
+        }.map { it.id }.toSet()
+    }
+
+    private fun removeNonexistentTablesAndColumnsForOrg(orgId: UUID, existingTableIds: Set<UUID>, existingColumnIds: Set<UUID>) {
+
+        // delete missing tables
+
+        val tableIdsToDelete = organizationExternalDatabaseTables.keySet(Predicates.equal(ORGANIZATION_ID_INDEX, orgId)).filter {
+            !existingTableIds.contains(it)
+        }.toSet()
+
+        if (tableIdsToDelete.isNotEmpty()) {
+            edms.deleteOrganizationExternalDatabaseTableObjects(tableIdsToDelete)
+        }
+
+
+        // delete missing columns
+
+        val columnIdsToDelete = organizationExternalDatabaseColumns.values(Predicates.equal(ORGANIZATION_ID_INDEX, orgId)).filter {
+            !existingColumnIds.contains(it.id)
+        }.groupBy { it.tableId }.mapValues { it.value.map { c -> c.id }.toSet() }
+
+        if (columnIdsToDelete.isNotEmpty()) {
+            edms.deleteOrganizationExternalDatabaseColumnObjects(columnIdsToDelete)
+        }
+    }
+
+    private fun createColumns(
+            table: OrganizationExternalDatabaseTable,
+            columns: List<OrganizationExternalDatabaseColumn>,
+            ownerAclKeys: List<AclKey>
+    ): List<OrganizationExternalDatabaseColumn> {
+        createSecurableColumnObjects(columns, ownerAclKeys, table)
+
+        return columns
+    }
+
+    private fun updateColumns(
+            columns: List<OrganizationExternalDatabaseColumn>,
+            existingColumnIdsByName: Map<String, UUID>
+    ): List<OrganizationExternalDatabaseColumn> {
+
+        val currentColumnsById = columns.associateBy { existingColumnIdsByName.getValue(it.getUniqueName()) }
+        val storedColumns = organizationExternalDatabaseColumns.getAll(existingColumnIdsByName.values.toSet())
+
+        val fullExistingColumns = mutableListOf<OrganizationExternalDatabaseColumn>()
+
+        val updatedColumns = storedColumns.mapNotNull { (id, storedColumn) ->
+            val currentColumn = currentColumnsById.getValue(id)
+            val updatedColumn = OrganizationExternalDatabaseColumn(
+                    id = storedColumn.id,
+                    name = currentColumn.name,
+                    title = storedColumn.title,
+                    description = Optional.of(storedColumn.description),
+                    tableId = currentColumn.tableId,
+                    organizationId = currentColumn.organizationId,
+                    dataType = currentColumn.dataType,
+                    primaryKey = currentColumn.primaryKey,
+                    ordinalPosition = currentColumn.ordinalPosition
+            )
+
+            fullExistingColumns.add(updatedColumn)
+
+            if (updatedColumn == storedColumn) null else updatedColumn
+        }.associateBy { it.id }
+
+        if (updatedColumns.isNotEmpty()) {
+            organizationExternalDatabaseColumns.putAll(updatedColumns)
+        }
```

Comment:
If `updatedColumns` is empty, would that mean that all the columns were removed? In which case, shouldn't hat also be reflected in the update?

Reply:
no, the cleanup not done in this function